File: programmers/old/level2/올바른_괄호.java
import java.io.*;
import java.util.*;
public class 올바른_괄호 {
	static
	class Solution {
	    boolean solution(String s) {
	        int top = -1;
	        for(int i = 0; i < s.length(); ++i) {
	        	char c = s.charAt(i);
	        	if(c == '(') {
	        		top++;
	        	}else {
	        		if(top >= 0) {
	        			top--;
	        		}else {
	        			return false;
	        		}
	        	}
	        }
	        if(top == -1)
	        	return true;
	        return false;
	    }
	}
	public static void main(String[] args) {
		String s = "()()";
		Solution sol = new Solution();
		System.out.println(sol.solution(s));
	}
}

File: programmers/js/new/level2/올바른_괄호.js
function solution(s) {
    let top = -1;
    const arr = s.split("");
    for (let i = 0; i < arr.length; ++i) {
        const c = arr[i];
        if (c === "(") {
            top++;
        } else {
            if (top < 0) {
                return false;
            } else {
                top--;
            }
        }
    }
    return top === -1;
}
